qdrant-landing/content/documentation/concepts/points.md

```diff
@@ -1632,12 +1631,97 @@ All resulting points are sorted by ID. To query the next page it is necessary to
+
+```http
+POST /collections/{collection_name}/points/scroll
+{
+    "limit": 15,
+    "order_by": "timestamp", // <-- this!
+}
+```
 
 ```python
+client.scroll(
+    collection_name="{collection_name}",
+    limit=15,
+    order_by="timestamp", # <-- this!
+)
 ```
- -->
+
+```typescript
+client.scroll("{collection_name}", {
+  limit: 15,
+  order_by: "timestamp", // <-- this!
+});
+```
+
+```rust
+use qdrant_client::qdrant::{Condition, Filter, ScrollPoints, OrderBy};
+
+client
+    .scroll(&ScrollPoints {
+        collection_name: "{collection_name}".to_string(),
+        limit: Some(15),
+        order_by: Some(OrderBy {
+            key: "timestamp".to_string(),  // <-- this!
+            ..Default::default(),
+        }),
+        ..Default::default()
+    })
+    .await?;
+```
+
+The `order_by` `key` parameter specifies the payload key to order the results by, but there are also other fields that can be set to control the ordering, like `direction` and `start_from`:
+
+```http
+"order_by": {
+    "key": "timestamp",
+    "direction": "desc" // default is "asc"
+    "start_from": 123, // start from this value
+}
+```
+
+```python
+order_by=models.OrderBy(
+    key="timestamp",
+    direction="desc",  # default is "asc"
+    start_from=123,  # start from this value
+)
+```
+
+```typescript
+order_by: {
+    key: "timestamp",
+    direction: "desc", // default is "asc"
+    start_from: 123, // start from this value
+}
+```
+
+```rust
+order_by: Some(OrderBy {
+    key: "timestamp".to_string(),
+    direction: Some(Direction::Desc) // default is Direction::Asc
+    start_from: Some(123),
+})
+```
+
+Be mindful that, for array-like payloads, each point can appear as many times as the number of elements in the array. For example, if you have a point payload with a `timestamp` key, and the value for the key is an array of 3 elements, the same point will appear 3 times in the results, one for each timestamp.
+
+<aside role="alert">Pagination is disabled when using the `order_by` parameter. </aside>
+
+As you may expect, pagination does not work exactly the same way than when the order is an ID, which is unique. Right now, pagination is not explicitly enabled when ordering by a payload key, so the `next_page_offset` field will not be present in the response. However, there is a trick you can do to achieve pagination:
+
+1. For each page, store the last value of the `order_by` field.
+2. Accumulate all ids which have the same "last value" in their `order_by` `key`.
+3. Make a `has_id` filter with all the ids you have accumulated.
+4. Use this filter and the "last value" as the `start_from` in the next request.
```

Comment:
Tip: As long as you've set up spaces correctly, you don't have to specify consecutive numbers. This makes life easier if  you have to insert / delete a step.

```suggestion
1. For each page, store the last value of the `order_by` field.
1. Accumulate all IDs with the same "last value" in their `order_by` `key`.
1. Create a `has_id` filter with accumulated IDs.
1. Use this filter and the "last value" as the `start_from` in the next request.
```

Reply:
Today I learned about this, thanks!